ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,277 @@
+  ClusterDeployments ---> |reference| cluster-ssh-key
+  ClusterDeployments ---> |reference| cluster-pull-secret
+  ClusterPools ---> |reference| install-configs
+  ClusterPools ---> |reference| aws-creds
+  ClusterPools ---> |reference| cluster-ssh-key
+  ClusterPools ---> |reference| cluster-pull-secret
+  ArgoCDOperator --> |installs| HelmCharts
+  ArgoCDOperator --> |watches| Applications
+  HiveOperator --> |watches| ClusterDeployments
+  HiveOperator --> |watches| ClusterClaims
+  HiveOperator --> |4. creates| kubeconfig
+  install-configs --> |contain copy of| cluster-pull-secret
+  install-configs --> |contain copy of| cluster-ssh-key
+  HelmCharts --> |3. deploy| ClusterDeployments
+  HelmCharts --> |3. deploy| ClusterClaims
+  ApplicationSets --> |reference| HelmCharts
+  ClusterClaims --> |reference| ClusterPools
+  IntegrationPipeline ---> |1. creates| ClusterTemplateInstances
+  IntegrationPipeline ---> |5. reads| kubeconfig
+  ClaaSOperator --> |2. creates| Applications
+  ClaaSOperator --> |watches| ClusterTemplateInstances
+```
+
+### Scalability
+
+The [CaaS] Operator should scale well. It hands off most workloads to ArgoCD. Even so,
+there are a few good reasons to deploy CaaS, Hive and/or Hypershift on clusters separate from the
+Konflux dataplane clusters:
+
+* Hive's upper limits, once reached, are ultimately only surmountable by adding more management
+  clusters.
+* The tight coupling of Hypershift management cluster version to provisionable control plane
+  versions suggests timely upgrades of the management cluster may be important to our end users.
+  The rest of the Konflux services on the dataplane may not support as aggressive of an upgrade
+  schedule.
+
+### Access Management
+  
+Introducing new cluster(s) creates complexity elsewhere. A tenant needs access to a namespace
+on the remote cluster within which they can manage select resources
+(e.g. `ClusterTemplateInstances`, `Secrets`, `ClusterPools`).
+
+We will either update or create [NSTemplateTiers] with the addition of a `SpaceRequest`. A new
+cluster role will be created on the `ToolchainCluster` Custom Resource to classify the cluster(s)
+used for test environment provisioning. The `SpaceRequest` controller, noticing the
+cluster role on the request, will create the namespace on one of the remote clusters. It
+will also create a secret in the tenant namespace containing credentials for a service account
+with access to the remote namespace. This secret can then be used from a `PipelineRun` workload
+like any other.
+
+The user will not be allowed to completely remove the `SpaceRequest` from their workspace as the
+member operator will restore it from the assigned `NSTemplateTier` if attempted.
+
+
+Should a new `NSTemplateTier` be created, existing tenants can be migrated to the new tier by an
+admin with a single `sandbox-cli` command. This technique can also be used for a manual approval
+workflow, if desired.
```

Comment:
When would this happen? Is there any concern with this related to users' ability to provision clusters?

Reply:
I'm expecting us to migrate all users and switch the default tier so this shouldn't be necessary. It's just an option should we need it.